examples/chef/common/clusters/oven-cavity-operational-state/chef-oven-cavity-operational-state.cpp

```diff
@@ -0,0 +1,317 @@
+        (void) DeviceLayer::SystemLayer().CancelTimer(onOvenCavityOperationalStateTimerTick, delegate);
+        return;
+    }
+
+    if (!delegate->CheckCycleActive())
+    {
+        (void) DeviceLayer::SystemLayer().CancelTimer(onOvenCavityOperationalStateTimerTick, delegate);
+        return;
+    }
+
+    delegate->CycleSecondTick();
+
+    if (delegate->CheckCycleComplete())
+    {
+        OperationalState::GenericOperationalError err(to_underlying(ErrorStateEnum::kNoError));
+        delegate->HandleStopStateCallback(err);
+    }
+    else
+    {
+        (void) DeviceLayer::SystemLayer().StartTimer(System::Clock::Seconds16(1), onOvenCavityOperationalStateTimerTick, delegate);
+    }
+}
+
+DataModel::Nullable<uint32_t> ChefDelegate::GetCountdownTime()
+{
+    if (mRunningTime.IsNull() || CheckCycleComplete())
+        return DataModel::NullNullable;
+
+    return DataModel::MakeNullable(kCycleSeconds - mRunningTime.Value());
+}
+
+bool ChefDelegate::StartCycle()
+{
+    if (!mRunningTime.IsNull())
+    {
+        ChipLogError(DeviceLayer, "StartCycle: Cycle is in progress. Can not start new cycle.");
+        return false;
+    }
+    mRunningTime.SetNonNull(static_cast<uint32_t>(0));
+    app::DataModel::Nullable<uint8_t> phase = GetRunningPhase();
+    if (phase.IsNull())
+    {
+        ChipLogError(DeviceLayer, "StartCycle: Phase did't update to non-NULL.");
+        return false;
+    }
+    if (phase.Value() != 0)
+    {
+        ChipLogError(DeviceLayer, "StartCycle: Unexpected start phase: %d.", phase.Value());
+        return false;
+    }
+    CHIP_ERROR err = GetInstance()->SetCurrentPhase(phase);
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(DeviceLayer, "StartCycle: Error setting current phase %" CHIP_ERROR_FORMAT, err.Format());
+    }
+    ChipLogDetail(DeviceLayer, "Oven Cycle Started. Current Phase: %d. Cycle time: %d", phase.Value(), kCycleSeconds);
+    return true;
+}
+
+void ChefDelegate::CycleSecondTick()
+{
+    if (mRunningTime.IsNull() || CheckCycleComplete())
+        return;
+    uint8_t opState                       = GetCurrentOperationalState();
+    DataModel::Nullable<uint8_t> oldPhase = GetRunningPhase();
+    if (opState == to_underlying(OperationalStateEnum::kRunning))
+    {
+        mRunningTime.SetNonNull(mRunningTime.Value() + 1);
+    }
+    else
+    {
+        ChipLogError(DeviceLayer, "CycleSecondTick: Invalid state %d", opState);
+        return;
+    }
+    DataModel::Nullable<uint8_t> newPhase = GetRunningPhase();
+    if (newPhase != oldPhase)
+    {
+        CHIP_ERROR err = GetInstance()->SetCurrentPhase(newPhase);
+        if (err != CHIP_NO_ERROR)
+        {
+            ChipLogError(DeviceLayer, "CycleSecondTick: Error setting current phase %" CHIP_ERROR_FORMAT, err.Format());
+        }
+    }
+    GetInstance()->UpdateCountdownTimeFromDelegate();
+
+    ChipLogDetail(DeviceLayer, "CycleSecondTick: Running time: %d.", mRunningTime.Value());
```

Comment:
```suggestion
    ChipLogDetail(DeviceLayer, "CycleSecondTick: Running time: %d.", mRunningTime.ValueOr(0));
```

I think your code is correct because we return if running time is null (or we ensure non-null), however maybe this is more resilient to any changes in code in the future.